st2client/st2client/commands/pack.py

```diff
@@ -0,0 +1,247 @@
+import st2client.utils.terminal as term
+
+
+LIVEACTION_STATUS_REQUESTED = 'requested'
+LIVEACTION_STATUS_SCHEDULED = 'scheduled'
+LIVEACTION_STATUS_DELAYED = 'delayed'
+LIVEACTION_STATUS_RUNNING = 'running'
+LIVEACTION_STATUS_SUCCEEDED = 'succeeded'
+LIVEACTION_STATUS_FAILED = 'failed'
+LIVEACTION_STATUS_TIMED_OUT = 'timeout'
+LIVEACTION_STATUS_ABANDONED = 'abandoned'
+LIVEACTION_STATUS_CANCELING = 'canceling'
+LIVEACTION_STATUS_CANCELED = 'canceled'
+
+LIVEACTION_COMPLETED_STATES = [
+    LIVEACTION_STATUS_SUCCEEDED,
+    LIVEACTION_STATUS_FAILED,
+    LIVEACTION_STATUS_TIMED_OUT,
+    LIVEACTION_STATUS_CANCELED,
+    LIVEACTION_STATUS_ABANDONED
+]
+
+
+class PackBranch(resource.ResourceBranch):
+    def __init__(self, description, app, subparsers, parent_parser=None):
+        super(PackBranch, self).__init__(
+            Pack, description, app, subparsers,
+            parent_parser=parent_parser,
+            read_only=True,
+            commands={
+                'list': PackListCommand,
+                'get': NoopCommand
+            })
+
+        self.commands['register'] = PackRegisterCommand(self.resource, self.app, self.subparsers)
+        self.commands['create'] = PackCreateCommand(self.resource, self.app, self.subparsers)
+        self.commands['install'] = PackInstallCommand(self.resource, self.app, self.subparsers)
+        self.commands['remove'] = PackRemoveCommand(self.resource, self.app, self.subparsers)
+        self.commands['search'] = PackSearchCommand(self.resource, self.app, self.subparsers)
+        self.commands['show'] = PackShowCommand(self.resource, self.app, self.subparsers)
+
+
+class PackResourceCommand(resource.ResourceCommand):
+    def run_and_print(self, args, **kwargs):
+        try:
+            instance = self.run(args, **kwargs)
+            if not instance:
+                raise resource.ResourceNotFoundError("No matching items found")
+            self.print_output(instance, table.PropertyValueTable,
+                              attributes=['all'], json=args.json, yaml=args.yaml)
+        except resource.ResourceNotFoundError:
+            print("No matching items found")
+        except Exception as e:
+            message = e.message or str(e)
+            print('ERROR: %s' % (message))
+            raise OperationFailureException(message)
+
+
+class PackAsyncCommand(ActionRunCommandMixin, resource.ResourceCommand):
+    def __init__(self, *args, **kwargs):
+        super(PackAsyncCommand, self).__init__(*args, **kwargs)
+
+        self._add_common_options()
+
+    @resource.add_auth_token_to_kwargs_from_cli
+    def run_and_print(self, args, **kwargs):
+        instance = self.run(args, **kwargs)
+        if not instance:
+            raise Exception('Server did not create instance.')
+
+        parent_id = instance.execution_id
+
+        stream_mgr = self.app.client.managers['Stream']
+
+        execution = None
+
+        with term.TaskIndicator() as indicator:
```

Comment:
Ah, cool I see how you solved the async API in the client.

This works for me :+1: